src/s2wasm.h

```diff
@@ -491,8 +494,11 @@ class S2WasmBuilder {
       skipWhitespace();
       if (match(".p2align")) s = strchr(s, '\n');
       else if (match(".int32")) {
-        initializerFunctions.emplace_back(cleanFunction(getStr()));
-        assert(implementedFunctions.count(initializerFunctions.back()));
+        do {
```

Comment:
Hm, I think the `while(*s)` is basically to catch the end of the buffer if it's terminated early. But in this function right now it actually doesn't do anything, because it either breaks or aborts. So we could remove the outer `while(*s)` and make the end of the inner loop be `} while (match(".int32") && *s);` (that will cover  the case where .int32 is the last thing in the buffer). Then there's no outer loop so no break.
 Also it seems like the p2align should be optional (although maybe in practice it will always be there?).